src/pages/waiting-room-to-car/cat-d/waiting-room-to-car.cat-d.page.ts

```diff
@@ -271,9 +271,11 @@ export class WaitingRoomToCarCatDPage extends BasePageComponent {
   onSubmit() {
     Object.keys(this.form.controls).forEach(controlName => this.form.controls[controlName].markAsDirty());
     if (this.form.valid) {
-      this.navController.push(CAT_D.TEST_REPORT_PAGE).then(() => {
+      this.navController.push(CAT_D.TEST_REPORT_PAGE).then(async () => {
         const view = this.navController.getViews().find(view => view.id === CAT_D.WAITING_ROOM_TO_CAR_PAGE);
-        if (view) {
+        let isDelegated: boolean;
+        await this.pageState.delegatedTest$.pipe(map(value => isDelegated = value)).subscribe();
```

Comment:
do we need these new subscriptions? remember subscriptions need to be unsubscribed to avoid memory leaks. if you wan to do like this you could use rxjs take in order to automatically unsubscribe after first emission of observable https://www.learnrxjs.io/learn-rxjs/operators/filtering/take

Reply:
thats a fair point, i will do them in a more standardised way more in line with the rest of the app, i wasnt aware of how important unsubscribing was, will be more cautious going forward, thanks matt